test/genesisProtocol.js

```diff
@@ -0,0 +1,393 @@
+
+describe("GenesisProtocol", () => {
+  let dao, genesisProtocol, paramsHash, executableTest;
+
+  const createProposal = async function () {
+
+    const result = await genesisProtocol.propose({
+      avatar: dao.avatar.address,
+      numOfChoices: 2,
+      paramsHash: paramsHash,
+      executable: executableTest.address
+    });
+
+    assert.isOk(result);
+    assert.isOk(result.proposalId);
+
+    const stakingToken = await DAOToken.at(await genesisProtocol.stakingToken());
+
+    await stakingToken.approve(accounts[0], web3.toWei(10));
+    await stakingToken.approve(genesisProtocol.address, web3.toWei(10));
+
+    return result.proposalId;
+  };
+
+  const voteProposal = function (proposalId, how) {
+    return genesisProtocol.vote({
+      proposalId: proposalId,
+      vote: how
+    });
+  };
+
+  const stakeProposalVote = function (proposalId, how, amount) {
+    return genesisProtocol.stake({
+      proposalId: proposalId,
+      vote: how,
+      amount: web3.toWei(amount)
+    });
+  };
+
+  beforeEach(async () => {
+
+    dao = await DAO.new({
+      name: "Skynet",
+      tokenName: "Tokens of skynet",
+      tokenSymbol: "SNT",
+      schemes: [
+        { name: "GenesisProtocol" }
+      ],
+      founders: [{
+        address: accounts[0],
+        reputation: web3.toWei(1000),
+        tokens: web3.toWei(1000)
+      }
+      ]
+    });
+
+    const scheme = await dao.getSchemes("GenesisProtocol");
+
+    assert.isOk(scheme);
+    assert.equal(scheme.length, 1);
+    assert.equal(scheme[0].name, "GenesisProtocol");
+
+    genesisProtocol = await GenesisProtocol.at(scheme[0].address);
+
+    assert.isOk(genesisProtocol);
+
+    // all default parameters
+    paramsHash = (await genesisProtocol.setParams({})).result;
+
+    executableTest = await ExecutableTest.deployed();
+  });
+
+
+  it("can call getScoreThresholdParams", async () => {
+    const result = await genesisProtocol.getScoreThresholdParams({
+      avatar: dao.avatar.address,
+    });
+    assert.isOk(result);
+    assert.equal(result.thresholdConstA, 1);
+    assert.equal(result.thresholdConstB, 1);
+  });
+
+  it("can call getShouldBoost", async () => {
+    const proposalId = await createProposal();
+    const result = await genesisProtocol.getShouldBoost({
+      proposalId: proposalId
+    });
+    assert.equal(result, false);
+  });
+
+
+  it("can call getVoteStake", async () => {
+    const proposalId = await createProposal();
+
+    await stakeProposalVote(proposalId, 1, 10);
+
+    const result = await genesisProtocol.getVoteStake({
+      proposalId: proposalId,
+      vote: 1
+    });
+    assert.equal(web3.fromWei(result), 10);
+  });
+
+  it("can call getVotesStatus", async () => {
+    const proposalId = await createProposal();
+    const result = await genesisProtocol.getVotesStatus({
+      proposalId: proposalId
+    });
+    assert.isOk(result);
+    // TODO: currently failing. Bug in Arc?  assert.equal(result.length, 2);
```

Comment:
If there is bug in ARC please open an issue for that.
If there is not remove this comment.
Remove commented code.

Reply:
I did create  an issue in Arc: https://github.com/daostack/arc/issues/328.

I am changing the comment in the code to reference that.

Reply:
There is no need to add a link to github issue at the code.
any how it is not a bug as it explained in the issue and it should not prevent test to pass

Reply:
The explanation made no sense to me, as I responded in the issue.  In any case, the question has to be resolved, and it appears it may be resolved because of the change making number of choices always be 2.

I think we need to be clear in the comments to the code in Arc that is it going to respond with MAX_NUM_OF_CHOICES **+ 1**

When that PR is merged, I'll grab it and address these lines of code.